### ISSUE TITLE
New Indonesian Translation Progress

### ISSUE THREAD

**Issue by gaearon** [reactjs/id.react.dev#330]
## Maintainer List

* @arfianadam 
* @gedeagas 
* @regalius 
* @rizafahmi 
* @resir014 
* @grikomsn

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the glossary and style guide (once they exist) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find that you can't commit anymore, let the maintainers know so they can assign the page to someone else.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

- [ ] Quick Start (@tesseralis) #12345

When PRs are merged, make sure to mark that page as completed!

## Main Content 

To do before releasing as an "official" translation. **Please translate these pages first.**

Note that each section has an index page, which needs to be translated too.

### Learn React

- [ ] Quick Start
  - [ ] Tutorial: Tic Tac Toe
  - [ ] Thinking in React
  
- [ ] Installation
  - [ ] Start a New React Project
  - [ ] Add React to an Existing Project
  - [ ] Editor Setup
  - [ ] React Developer Tools
  
- [ ] Describing the UI
  - [ ] Your First Component
  - [ ] Importing and Exporting Components
  - [ ] Writing Markup with JSX
  - [ ] JavaScript in JSX with Curly Braces
  - [ ] Passing Props to a Component
  - [ ] Conditional Rendering
  - [ ] Rendering Lists
  - [ ] Keeping Components Pure
  
- [ ] Adding Interactivity
  - [ ] Responding to Events
  - [ ] State: A Component's Memory
  - [ ] Render and Commit
  - [ ] State as a Snapshot
  - [ ] Queueing a Series of State Updates
  - [ ] Updating Objects in State
  - [ ] Updating Arrays in State
  
- [ ] Managing State
  - [ ] Reacting to Input with State
  - [ ] Choosing the State Structure
  - [ ] Sharing State Between Components
  - [ ] Preserving and Resetting State
  - [ ] Extracting State Logic into a Reducer
  - [ ] Passing Data Deeply with Context
  - [ ] Scaling Up with Reducer and Context
  
- [ ] Escape Hatches
  - [ ] Referencing Values with Refs
  - [ ] Manipulating the DOM with Refs
  - [ ] Synchronizing with Effects
  - [ ] You Might Not Need an Effect
  - [ ] Lifecycle of Reactive Effects
  - [ ] Separating Events from Effects
  - [ ] Removing Effect Dependencies
  - [ ] Reusing Logic with Custom Hooks

### API Reference

- [ ] `react`: Hooks
  - [ ] `useCallback`
  - [ ] `useContext`
  - [ ] `useDebugValue`
  - [ ] `useDeferredValue`
  - [ ] `useEffect`
  - [ ] `useId`
  - [ ] `useImperativeHandle`
  - [ ] `useInsertionEffect`
  - [ ] `useLayoutEffect`
  - [ ] `useMemo`
  - [ ] `useReducer`
  - [ ] `useRef`
  - [ ] `useState`
  - [ ] `useSyncExternalStore`
  - [ ] `useTransition`

- [ ] `react`: Components
  - [ ] `<Fragment> (<>)`
  - [ ] `<Profiler>`
  - [ ] `<StrictMode>`
  - [ ] `<Suspense>`

- [ ] `react`: APIs
  - [ ] `createContext`
  - [ ] `forwardRef`
  - [ ] `lazy`
  - [ ] `memo`
  - [ ] `startTransition`

- [ ] `react-dom`: Components
  - [ ] Common (e.g. `<div>`)
  - [ ] `<input>`
  - [ ] `<option>`
  - [ ] `<progress>`
  - [ ] `<select>`
  - [ ] `<textarea>`

- [ ] `react-dom`: APIs
  - [ ] `createPortal`
  - [ ] `flushSync`
  - [ ] `findDOMNode`
  - [ ] `hydrate`
  - [ ] `render`
  - [ ] `unmountComponentAtNode`

- [ ] `react-dom/client`: Client APIs
  - [ ] `createRoot`
  - [ ] `hydrateRoot`

- [ ] `react-dom/server`: Server APIs
  - [ ] `renderToNodeStream`
  - [ ] `renderToPipeableStream`
  - [ ] `renderToReadableStream`
  - [ ] `renderToStaticMarkup`
  - [ ] `renderToStaticNodeStream`
  - [ ] `renderToString`

### Navigation and UI

We suggest to leave *most* of the UI translation until the end. We plan to do some invasive changes to the website folder layout and components, so postponing this until your translation is almost complete would make it easier to merge the changes from our side later. It might make sense to translate the homepage above the fold early, but leave the rest for later. As individual pages get translated, you can change the page titles in the corresponding sidebar files. Finally, when you're translating the navigation, make sure to test both desktop and mobile layouts.

- [ ] Homepage (currently in `HomeContent.js`)
- [ ] Sidebars (currently in `src/sidebar*.json`)
- [ ] Top-level navigation (currently in `TopNav.tsx`)

### When You're Ready...

After everything above is translated, add your language to `deployedLanguages` in `Seo.tsx` of the original [reactjs/react.dev](https://github.com/reactjs/react.dev) repository.

## Secondary Content 

These API pages should ideally be translated too, but they're less urgent and can be done after the others:

- [ ] Legacy React APIs
  - [ ] `Children`
  - [ ] `cloneElement`
  - [ ] `Component`
  - [ ] `createElement`
  - [ ] `createFactory`
  - [ ] `createRef`
  - [ ] `isValidElement`
  - [ ] `PureComponent`

## Optional Content 

These aren't the main translation targets, but if you'd like to do them, feel free to expand the list to include their subpages:

- [ ] Community
- [ ] Blog
- [ ] Warnings

**Comment by infrasync** [comment]
May i translate a Quick Start Page ?

**Comment by mustafamilyas** [comment]
I'm interested in translating, "describing the UI" section, I can start until your first component today, may I?

**Comment by syauqy** [comment]
Hi, I'm interested to translate "Start a New React Project" under the Installation section.

**Comment by rickyraz** [comment]
I'm interested in contributing, may I translate the page 'Conditional Rendering'

**Comment by padunk** [comment]
Hi, love to contribute back to React community, I would like to translate Add React to an Existing Project

**Comment by chrsep** [comment]
Hi! I'd also love to contribute back to react. Can i take the **Adding Interactivity** overview page?

**Comment by kalmndo** [comment]
I'm really loved if i can contribute on **Managing State**

**Comment by samnap11** [comment]
I would like to contribute as well by translating 2 pages, which are "Rendering Lists" and "Keeping Components Pure".

**Comment by mhaidarhanif** [comment]
Hi @resir014 and @grikomsn , once you've confirmed the question from @gaearon, we three @zainfathoni @r17x @mhaidarhanif also would like to be listed as well. Thank you

**Comment by resir014** [comment]
> Can you please confirm if you will be able to maintain and steer this translation? We need to get [reactjs/translations.react.dev@main/langs/id.json](https://github.com/reactjs/translations.react.dev/blob/main/langs/id.json?rgh-link-date=2023-04-25T15%3A17%3A19Z) to match the current list of active maintainers. Thank you!

Hi @gaearon!

Sorry for the late response, but yeses, I'll be able to manage the translation for for the Indonesian localisation. And as @mhaidarhanif mentioned, we can add him, @r17x, and @zainfathoni to the maintainers list.

**Comment by resir014** [comment]
> May i translate a Quick Start Page ?

@infrasync Of course! Your assigned page will be: **Tutorial: Tic Tac Toe**

As for me, I'll take the other one: **Thinking in React**.

**Comment by resir014** [comment]
> I'm interested in translating, "describing the UI" overview page, may I?

@mustafamilyas Of course!

Your next page will be: **Describing the UI**

**Comment by resir014** [comment]
> Hi, I'm interested to translate "Installation" and "Start a New React Project" page under the Installation section.

@syauqy You got it!

Your next page(s) will be: **Installation**, *Start a New React Project**

**Comment by resir014** [comment]
> I would like to translate the "Editor Setup" page under the Installation section

@agprst Your next page will be: **Editor Setup**

**Comment by resir014** [comment]
> I'm interested in contributing, may I translate the page 'Conditional Rendering'

@rickyraz Your next page will be: **Conditional Rendering**

> Hi, love to contribute back to React community, I would like to translate Add React to an Existing Project

@padunk Your next page will be: **Add React to an Existing Project**

> Hi! I'd also love to contribute back to react. Can i take the Adding Interactivity overview page?

@chrsep Your next page will be: **Adding Interactivity**

> I'm really loved if i can contribute on Managing State

@kalmndo Your next page will be: **Managing State**

> I would like to contribute as well by translating "Rendering Lists".

@samnap11 Your next page will be: **Rendering Lists**

Thank you all!

**Comment by mhaidarhanif** [comment]
@resir014 I noticed that "Quick Start" hasn't been assigned yet

URL: https://react.dev/learn

If still available, and if I may, I'll take it

**Comment by andriawan** [comment]
Hi maintaner @resir014 @mhaidarhanif  @r17x , i would like to help translating this page https://react.dev/learn/adding-interactivity

**Comment by yaffalhakim1** [comment]
Hi, I'm interested in contributing. may I translate the `React Developer Tools?`

**Comment by resir014** [comment]
> Hi maintaner, i would like to help translating this page [react.dev/learn/adding-interactivity](https://react.dev/learn/adding-interactivity)

Hi @andriawan - that page has already been claimed. Would you like to pick another page?

> Hi, I'm interested in contributing. may I translate the React Developer Tools?

@yaffalhakim1 All set!

Your next page will be: **React Developer Tools**

**Comment by mhmmd-ysf** [comment]
I assume `Writing Markup with JSX` hasn't been claimed, if it's not then i'll start there.

**Comment by andriawan** [comment]
> > Hi maintaner, i would like to help translating this page [react.dev/learn/adding-interactivity](https://react.dev/learn/adding-interactivity)
> 
> Hi @andriawan - that page has already been claimed. Would you like to pick another page?

sure @resir014. what about `Manipulating the DOM with Refs` https://react.dev/learn/manipulating-the-dom-with-refs

**Comment by samnap11** [comment]
@resir014 In the beginning of each chapter, there is a `<YouWillLearn>` component, e.g. `Your First Component` page. The block starts with the title of **You Will Learn**. I'm having trouble translating the title because the title is already hard-coded into the docs using English. What is the solution for this?  https://github.com/reactjs/id.react.dev/blob/57d8f80d032a45ff0bac69376b6442254cf98ff2/src/components/MDX/MDXComponents.tsx#L169-L178

**Comment by resir014** [comment]
> Hi, I'm interested to translate react: APIs -> createContext

@r17x Okay. Your next page will be: **`react`: APIs -> `createContext`** and **`react`: Hooks -> `useContext`**

**Comment by resir014** [comment]
> In the beginning of each chapter, there is a <YouWillLearn> component, e.g. Your First Component page. The block starts with the title of You Will Learn. I'm having trouble translating the title because the title is already hard-coded into the docs using English. What is the solution for this?

@samnap11 We can worry about localisation codes in layout component later, after all the docs contents are finished (i.e. everything in `.md`) For now, let's focus on the docs contents.

**Comment by resir014** [comment]
> sure @resir014. what about Manipulating the DOM with Refs [react.dev/learn/manipulating-the-dom-with-refs](https://react.dev/learn/manipulating-the-dom-with-refs) ?

@andriawan Sure!

Your next page will be: **Manipulating the DOM with Refs**

**Comment by AudiWu** [comment]
Hi, I'm interested in contributing. may I translate the react components fragment?

**Comment by chez14** [comment]
Hi @resir014 / other maintainer team, I'd like to translate [`react-dom/client` → `createRoot`](https://react.dev/reference/react-dom/client/createRoot) please. Thanks!

**Comment by naufaldi** [comment]
Hi @resir014 / @grikomsn ,  I'm interested in contributing. may I translate JavaScript in JSX with Curly Braces

**Comment by nagacoder** [comment]
Hi @resir014  / @grikomsn  , I'm interested in contributing in `Warnings` for transalation. Thanks

**Comment by DevinWinando** [comment]
Hello! I would love to translating `Passing Props to a Component`  if it hasn't been claimed.

**Comment by aroyan** [comment]
Hello @resir014, Can I contribute to `Importing and Exporting Components` ?

**Comment by resir014** [comment]
Also, @aroyan @DevinWinando @naufaldi @chez14 you all have been assigned to your pages!

**Comment by nagacoder** [comment]
@resir014  - i see, i will take `react:APIs -> memo ` then

**Comment by AudiWu** [comment]
> Also, @aroyan @DevinWinando @naufaldi @chez14 you all have been assigned to your pages!

Hi @resir014 I didn't see the part that I would take assigned to me.
`react: components <Fragment> (<>)`
Just to remind you thank you

**Comment by resir014** [comment]
@AudiWu Sorry, I missed it!

Your next page will be: **`react`: Components -> `<Fragment>`**

**Comment by resir014** [comment]
> i see, i will take react:APIs -> memo  then

@nagacoder Your next page will be: **`react`: APIs -> `memo`**

**Comment by arfianadam** [comment]
@gaearon hey, yes, sure :+1:

**Comment by KevinPalowa** [comment]
Hi, I'd love to  contribute on page `react`: Hooks

**Comment by samnap11** [comment]
@resir014 I just finished **Rendering Lists** page. I would like to help with the next page, which is **Keeping Components Pure** while waiting for the review.

**Comment by junkevin88** [comment]
Hi, I would love to contribute on page `react-dom/client`: Clients APIs-> `hydrateRoot`

**Comment by dimassrio** [comment]
@resir014 Would love to edit `Responding to Events` page.

**Comment by dimasma0305** [comment]
Hello, @resir014. I would like to request to translate the `react-dom: APIs` page. Could you please assign it to me?

**Comment by aroyan** [comment]
Hello @resir014, Can I contribute to `react-dom/Components/option` next?

**Comment by averrows** [comment]
hi @resir014, I'd like to contribute to `You Might Not Need an Effect`. can you assign it to me? thank you

**Comment by JfrAziz** [comment]
Hello @resir014 , can I contribute to `Managing State > Reacting to Input with State`?

**Comment by milhamm** [comment]
Hi @resir014. I would like to contribute to `useLayoutEffect`

**Comment by intanirnanda5796** [comment]
hi @resir014, can i contribute to `useCallback` ?

**Comment by vhermawan** [comment]
hi @resir014, i'm interesting to contribute. May i help to translate  `Render and Commit` ?

**Comment by resir014** [comment]
> I saw that i was assign to react-dom: Components -> Common (e.g. <div>) ?

@AudiWu Sotty! You have the correct page now.

**Comment by ilham89** [comment]
hi @resir014 , can I contribute to `API Reference > react-dom > Component > progress`?

**Comment by resir014** [comment]
Also, morning everyone! I've sorted out through all requests for page claims, and you should all be assigned to the correct pages. If something is wrong, let me know!

**Comment by dzakiafif** [comment]
hi @resir014, can i contribute to `Managing State > Choosing the State Structure` ?

**Comment by krowter** [comment]
Hi @resir014. I'm interested in translating `Adding Interactivity > State: A Component's Memory`. Thank you

**Comment by xevorte** [comment]
Hi @resir014, may i contribute to  `API Reference > Hooks > useRef` ? Thanks!

**Comment by brandon-julio-t** [comment]
Hi, @resir014, I'm interested in translating `Updating Objects in State`.? Thank you.

**Comment by ridhlab** [comment]
Hi, @resir014, I'm interested in translating `Passing Data Deeply with Context`. May I contribute? Thank you.

**Comment by lionelritchie29** [comment]
Hi, @resir014 , I am interested in translating `State as a Snapshot`. May I contribute to the translation ? Thank you

**Comment by AndriP8** [comment]
Hi @resir014 , I'm interested to contribute, may I translate the `Hooks > useId` page ?

**Comment by R1dwanMaulana** [comment]
Hello @resir014 , can I contribute to Managing State > Choosing the State Stucture? thanks

**Comment by rvndaputra** [comment]
hi @resir014, I'd like to contribute to the translation of `Reusing Logic with Custom Hooks`. may i help with the translation? thank you

**Comment by yehezkielgunawan** [comment]
Hi mas @resir014, I'd like to contribute to translating the `react-dom: Components` overview page. May I?

Thank you.

**Comment by R1dwanMaulana** [comment]
Hello @resir014 , can I contribute to useState? thanks

**Comment by rimuhamu** [comment]
Hi @resir014, I would like to contribute to translate `react: Components -> <Profiler>` page.

Thanks.

**Comment by dhafitf** [comment]
Hi @resir014. I would like to contribute to `Updating Arrays in State`. Thanks.

**Comment by resir014** [comment]
I've done the second round of assigning pages for those who already claimed them above. Happy translating!

**Comment by r17x** [comment]
@resir014 Need update for task `forwardRef` & `useContext` (assignee to me & linked PRs)

**Comment by annisann** [comment]
Hi @resir014, I'd like to contribute to `useMemo`. Thanks! :)

**Comment by rimuhamu** [comment]
@resir014 I think you assign the wrong page to @dhafitf

**Comment by yaffalhakim1** [comment]
Hi @resir014 is it ok if i do more translation? if yes i would like to translate `Sharing State Between Components`. Thank you

**Comment by mrevanzak** [comment]
hi @resir014, can I contribute to  `Queueing a Series of State Updates`

**Comment by ramadanriz** [comment]
Hi @resir014 I want to contribute to react-dom component "<select>". Thank You

**Comment by rivldy** [comment]
Hi @resir014, I'd like to contribute translating the `useReducer` page in the references section please.

**Comment by hendraaagil** [comment]
Hi @resir014. I would like to contribute to `Managing State > Preserving and Resetting State`. Thanks.

**Comment by lauslim12** [comment]
Good evening @resir014, I would like to contribute for `<StrictMode />`. Thank you!

**Comment by dickygiancini** [comment]
Hi @resir014 , right now I'm still learning react but I would love to help and contribute translating `useTransition` , is it possible for me? Thank you in advance!

**Comment by FelixFern** [comment]
Hello @resir014, I really would love to contribute to this project, can I help in translating the `Sharing State Between Components` page? Thank you in advance!

**Comment by maulanasdqn** [comment]
I would like to translate the "Suspense" page under the React Component section

**Comment by milhamm** [comment]
Hi @resir014 I would like to contribute also to `react-dom/client: Client APIs` page

**Comment by zaynhakim** [comment]
Hi @resir014 , I would like to contribute on `renderToPipeableStream` page

**Comment by mhaidarhanif** [comment]
@resir014 We need to confirm progress with PR for these because some are approaching/near 3 days elapsed after the initial assignment.

Also want to ask why some of the assigned are not in the comments (?).

I've already rechecked the open and closed PR lists.

### Learn React (High Priority)

- Quick Start (@infrasync)
- Add React to an Existing Project (@padunk)
- Writing Markup with JSX (@mhmmd-ysf)
- JavaScript in JSX with Curly Braces (@naufaldi)
- Passing Props to a Component (@DevinWinando)
- Conditional Rendering (@rickyraz)
- Keeping Components Pure (@samnap11)
- Responding to Events (@dimassrio)
- State: A Component's Memory (@krowter)
- Updating Objects in State (@brandon-julio-t)
- Managing State (@kalmndo)
- Reacting to Input with State (@JfrAziz)
- Choosing the State Structure (@dzakiafif)
- Extracting State Logic into a Reducer (@thaddeuscleo)
- Manipulating the DOM with Refs (@andriawan)
- You Might Not Need an Effect (@averrows)

### API Reference (High Priority)

- `react`: Hooks (@KevinPalowa)
- `useCallback` (@intanirnanda5796)
- `useDebugValue` (@arganaphang)
- `<Fragment> (<>)` (@AudiWu)
- `memo` (@nagacoder)
- `<option>` (@aroyan)
- `react-dom`: APIs (@dimasma0305)
- `createRoot` (@chez14)
- `hydrateRoot` (@junkevin88)

**Comment by irsyaadbp** [comment]
Hi @resir014, I'd like to contribute to translating the `Referencing Values with Refs` page

**Comment by Safiqq** [comment]
Hi @resir014 I'd like to contribute to translate the `react-dom/server`: Server APIs

**Comment by ceavinr** [comment]
Hi @resir014 I'd like to contribute to translate the `renderToStaticMarkup` page

**Comment by resir014** [comment]
> The goal is to make sure the availability of existing assignees and unblock new translators who want to take the pages. Also to make sure we follow the guideline in the original post by gaearon: Please take only one page at a time.

@mhaidarhanif Yep, that's what I'm doing. ATM I'm letting them take a new one once they have a PR ready.

**Comment by hilmibaskara** [comment]
Hi @resir014 I'd like to contribute to translate to `renderToStaticNodeStream`  page

**Comment by adamcanray** [comment]
Hi @resir014, may I translate the`useDeferredValue` page?

**Comment by resir014** [comment]
> Hello @resir014, I really would love to contribute to this project, can I help in translating the Sharing State Between Components page? if it's assigned to someone else, can i help in translating the Preserving and Resetting State  page? Thank you in advance!

@FelixFern Hi, those two pages have been claimed. Do you mind claiming another page?

**Comment by resir014** [comment]
> Hi @resir014 , i would like to contribute on Sharing State Between Components page

Hi @shaddamalbz, that page has been claimed. Would you like to claim another page?

**Comment by FelixFern** [comment]
> > Hello @resir014, I really would love to contribute to this project, can I help in translating the Sharing State Between Components page? if it's assigned to someone else, can i help in translating the Preserving and Resetting State  page? Thank you in advance!
> 
> @FelixFern Hi, those two pages have been claimed. Do you mind claiming another page?

Thanks for the reply! yes I wouldn't mind at all, how about the `Referencing Values with Refs` page? is it claimed yet?

**Comment by resir014** [comment]
> > > Hello @resir014, I really would love to contribute to this project, can I help in translating the Sharing State Between Components page? if it's assigned to someone else, can i help in translating the Preserving and Resetting State  page? Thank you in advance!
> > 
> > 
> > @FelixFern Hi, those two pages have been claimed. Do you mind claiming another page?
> 
> Thanks for the reply! yes I wouldn't mind at all, how about the `Referencing Values with Refs` page? is it claimed yet?

@FelixFern Yes, you can!

Your next page will be: **Referencing Values with Refs**

**Comment by mhaidarhanif** [comment]
@resir014 @FelixFern `Referencing Values with Refs` was requested by @irsyaadbp 2 hours ago. Therefore @irsyaadbp have to request another one.

**Comment by FelixFern** [comment]
> > > > Hello @resir014, I really would love to contribute to this project, can I help in translating the Sharing State Between Components page? if it's assigned to someone else, can i help in translating the Preserving and Resetting State  page? Thank you in advance!
> > > 
> > > 
> > > @FelixFern Hi, those two pages have been claimed. Do you mind claiming another page?
> > 
> > Thanks for the reply! yes I wouldn't mind at all, how about the `Referencing Values with Refs` page? is it claimed yet?
> 
> @FelixFern ~~Yes, you can!~~
> 
> ~~Your next page will be: **Referencing Values with Refs**~~ Nevermind, see below

oh ok no probs! in that case, how about the 'Common (e.g. <div>)' page?

**Comment by samnap11** [comment]
@mhaidarhanif Would you prefer us to open a draft PR as soon as possible or open it only when we're ready to merge it? Or perhaps your point was to simply push us to be better and quicker at translating the pages?

**Comment by mhaidarhanif** [comment]
> @FelixFern oh ok no probs! in that case, how about the 'Common (e.g. div)' page?

Yes, that page `Common (e.g. <div>)` hasn't been requested/claimed, so it's safe (to be taken by @FelixFern). Will wait for @resir014 to assign.

Therefore @irsyaadbp can still take `Referencing Values with Refs` as he requested earlier.

**Comment by samnap11** [comment]
Also, I would like to suggest that maybe we can use a spreadsheet to take assignments. Because, currently, it is very chaotic and difficult to maintain the requests for assignments from comments only due to how fast the number of comments is increasing, at least from my perspective.

**Comment by mhaidarhanif** [comment]
@samnap11 Agree, it seems confusing. Also about this, need some input from @resir014 @mazipan @r17x @zainfathoni 

Initially, there was an idea to make a GitHub Projects board (instead of a spreadsheet). We'll see after some discussions.

**Comment by arfianadam** [comment]
> as we've been discussing it internally

@mhaidarhanif Hello, what kind of internal discussions are we talking about? Can we use a public-accessible channel instead of private one? Can be just "Discussions" in this repo, IMO.

**Comment by mhaidarhanif** [comment]
> > as we've been discussing it internally
> 
> @mhaidarhanif Hello, what kind of internal discussions are we talking about? Can we use a public-accessible channel instead of private one? Can be just "Discussions" in this repo, IMO.

Sure, we'll wait for @resir014 @r17x @zainfathoni to confirm to move forward about this to this issue and "Discussions".

**Comment by averrows** [comment]
Hi @resir014, I'd like to contribute to `react: Hooks > useSyncExternalStore`. Can you please assign it to me? thank you.

**Comment by juliansyahrifqi** [comment]
Hello, I am interested to translate `react-dom pages: Components > <textarea>` page. Thank you

**Comment by shaddamalbz** [comment]
> Hi @shaddamalbz, that page has been claimed. Would you like to claim another page?

Hi @resir014, sure, how about `<StrictMode>` is it have been claimed yet?

**Comment by abidra** [comment]
Hi @resir014, I'd like to contribute to react: react-dom > textarea. Can you please assign it to me? thank you.

**Comment by fandok** [comment]
Hi @resir014, I would like to contribute to `Escape Hatches` page. Can you help to assign the page to me? Thank you very much!

**Comment by stackpan** [comment]
Can i contribute to `Escape Hatches` > `Synchronizing with Effects`?

**Comment by andreepratama27** [comment]
Hi, Can I contribute and assign me to page `Escape Hatches` > `Removing Effect Dependencies`, please?

**Comment by resir014** [comment]
Hey everyone!

Thank you for your enthusiasm in this translation project. To anticipate the demand, we're currently updating the workflow of our translation project. We're converting the checklist on the [original post](https://github.com/reactjs/id.react.dev/issues/330#issue-1682333016) into sub-issues so that everyone can claim them directly from said issue, rather than polluting this issue thread.

As soon as our work is done, this issue thread can be used for general discussions related to the translation project.

Thanks!

**Comment by doaortu** [comment]
> 1. Having all the issues ready for unfinished translations, because there are about 100 of them.

Hey @mhaidarhanif, I can help opening those issue for you.

**Comment by resir014** [comment]
> Hey @mhaidarhanif, I can help opening those issue for you.

@doaortu The maintainers are running through all the issues one by one. You can relax for now.

**Comment by doaortu** [comment]
> @doaortu The maintainers are running through all the issues one by one. You can relax for now.

Ah I see, thank you for the update.

**Comment by ical10** [comment]
Hi, I'm interested in contributing. Where can I see pages that are not assigned yet?

**Comment by mhaidarhanif** [comment]
@ical10 Please wait, as explained by @resir014 above, we're migrating to a better workflow so "we can see pages that are not assigned yet" much easier.

**Comment by zainfathoni** [comment]
As some of us suggested, we are mechanism to avoid unnecessary bottlenecks during the assignment process. The process is inspired by [WargaBantuWarga](https://github.com/kawalcovid19/wargabantuwarga.com/blob/main/CONTRIBUTING.md#issue-assignment--communication) project.

<details>
<summary>Here's the gist of the process, we will move it to the corresponding documentation soon.</summary>

#### Issue assignment & Communication

Once you're ready with your branch and have something to contribute, you'll want to
let everyone know that you are working on the issue. To communicate this, we
are using GitHub's Draft Pull Requests.

Draft Pull Request is like a regular Pull Request but it can't be merged until
it's marked as "ready for review". It signals other contributors that [it's a
work in progress](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request).
This is necessary to signal other contributors that the work for the particular
issue has started and it is still in progress. Also, it is a better approach to
use as a communication tool between contributors because we can provide
additional information other than viewing the changed files.

Therefore, when you have at least one commit **it's important to create a Draft
Pull Request** to let everyone know that the issue is assigned to you.

#### Creating a Draft Pull Request

Steps to creating a Draft Pull Request:

1. Commit and push your new changes into the remote repository. Please refer to [@commitlint/config-conventional](https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-conventional) for your commit message or you can use [commitlint.io](https://commitlint.io/) to assist you in composing the commit message.
2. Head over to the Pull requests section on your forked repository, hit New pull request.
   ![Hint-1](https://user-images.githubusercontent.com/46013258/126284390-c2bd1aa6-fdc2-4aa6-a945-031f02db038e.png)
3. Pick your forked repository for the head repository, and compare with the branch that you are having changes in.
   ![Hint-2](https://user-images.githubusercontent.com/46013258/126285036-27b49325-62a2-4a6c-b216-5bae261788da.png)
4. Put a clear title and description in your pull request. Make sure the
   description follows [our guide below](#formatting-pull-request-description).
   ![Hint-3](https://user-images.githubusercontent.com/46013258/126286179-04341e30-1224-49cb-9b9a-3c3aee99c308.png)
5. Pick Create draft pull request (like in the image above) and hit the green button.
6. Don't forget to mark your Draft Pull Request as Ready for review after you commit all of the changes.

#### Formatting Pull Request Description

To properly [link a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue), there is one tiny requirement text to put in a Pull Request description.
Please make sure to mention the issue that you're working on correctly. Replace
this text `<!-- mention the issue that you're trying to close with this PR -->`
from the template with the issue number. Example:

```markdown
Closes #318
```
</details>

**Comment by zainfathoni** [comment]
I updated the [TRANSLATION.md](https://github.com/reactjs/id.react.dev/blob/main/TRANSLATION.md) page with the abovementioned mechanism. Please take a look at that new process from now on.

Thank you for contributing, and happy translating! 😄

**Comment by zainfathoni** [comment]
In principle, anyone available to translate any page, please raise a Draft PR at your earliest convenience.
The issue description's `Assignee: @username` text is a remnant of the previous assignment process.
From now on, we will prioritise any PRs (or draft PRs) as a sign of self-assignment to the particular issue.

**Comment by resir014** [comment]
Hello everyone:

I'm pleased to announced that we've reached **50% progress** in our Core translations! 🎉 

<img width="353" alt="image" src="https://github.com/reactjs/id.react.dev/assets/5663877/f8e9746d-4c7c-47ec-91cd-c8532fa04d8e">

With this milestone, we've also announced that we're also opening translations for the medium priority issues, starting from the [Legacy React APIs](https://id.react.dev/reference/react/legacy) section. Please check the [first post](https://github.com/reactjs/id.react.dev/issues/330#issue-1682333016) for links to these issues.

Additionally, we're also opening translations for the UI elements once we got all the translated document titles figured out - since there are files like `sidebar.json` which relies on said translated document titles.

Thank you for your continued contribution! 💙

**Comment by resir014** [comment]
We've added some new pages from the upstream version to translate. Also there's a new "React Reference Overview" and "Rules of React section available to translate here: #649

**Comment by resir014** [comment]
Wave 2 translations for `id.react.dev` have been opened!

In this second wave, we will translate new pages added in anticipation of the new features to be added in React 19, as well as features in Canary channel, and also React Server Components.

![image](https://github.com/reactjs/id.react.dev/assets/5663877/6453d59e-d051-44c6-890d-6265350bb295)

![image](https://github.com/reactjs/id.react.dev/assets/5663877/912ea56f-0234-4a46-b10c-4b5df8b1acaa)

Please claim your issue(s) here: #681
